streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java

```diff
@@ -53,6 +53,11 @@ public KTableRepartitionMap(KTableImpl<K, ?, V> parent, KeyValueMapper<K, V, Key
             public KTableValueGetter<K, KeyValue<K1, V1>> get() {
                 return new KTableMapValueGetter(parentValueGetterSupplier.get());
             }
+
+            @Override
+            public String[] storeNames() {
+                return parentValueGetterSupplier.storeNames();
```

Comment:
I'd suggest throwing an runtime exception since the `storeNames` function should never be called for `KTableRepartitionMap`: it is more efficient for detecting errors.